src/main/scala/com/zeotap/data/io/source/spark/constructs/DataFrameOps.scala

```diff
@@ -71,6 +76,52 @@ object DataFrameOps {
 
       dataFrame.withColumn("inputPathColumn", addInputPathColumn(input_file_name())).select("inputPathColumn").distinct().collect.map(row => row.getString(0))
     }
+
+    /**
+     * Partitions given dataframe into number of partitions provided and writes the
+     * partitioned dataframe to the intermediate path provided, so that parallel processing can take place.
```

Comment:
Comment need to explicitly mention that split data, if already present, loads from ad-hoc path.